src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,464 @@
+    public ReadBaseStratification.Stratifiers STRATIFIER_VALUE = null;
+
+    @Argument(shortName = "V", doc = "VCF of loci to avoid collecting data (should include any known variation!)")
+    public File VCF=null;
+
+    @Argument(shortName = "L", doc = "Region(s) to limit analysis to. Can be given as a interval_list file. (will intersect if multiple values are given.) ", optional = true)
+    public List<File> INTERVALS = null;
+
+    @Argument(shortName = StandardOptionDefinitions.MINIMUM_MAPPING_QUALITY_SHORT_NAME, doc = "Minimum Mapping quality to include read.", minValue = 0)
+    public int MIN_MAPPING_Q = 20;
+
+    @Argument(shortName = "BQ", doc = "Minimum Base quality to include base.",minValue = 0)
+    public int MIN_BASE_Q = 20;
+
+    @Argument(shortName = "PE", doc = "The prior error, in phred-scale (used for calculating ratios and filtering out HET sites)",
+            optional = true, minValue = 2)
+    public int PRIOR_Q = 30;
+
+    @Argument(shortName = "MAX", doc = "Maximum number of loci to process (or unlimited if 0)", optional = true, minValue = 0)
+    public long MAX_LOCI = 0;
+
+    @Argument(shortName = "LH", doc = "Shortest homopolymer which is considered long.  Used by the BINNED_HOMOPOLYMER stratifier.", optional = true)
+    public int LONG_HOMOPOLYMER = 6;
+
+    @Argument(shortName = "P", doc = "The probability of selecting a locus for analysis (for downsampling)", optional = true, maxValue = 1, minValue = 0)
+    public double PROBABILITY = 1.0d;
+
+    @Override
+    protected boolean requiresReference() {
+        return true;
+    }
+
+    @Override
+    protected String[] customCommandLineValidation() {
+        List<String> errors = new ArrayList<>();
+
+        if (ERRORS_VALUE != null) {
+            errors.add("ERRORS_VALUE is a fake argument that is only there to show what are the different Error aggregation options. Please use it within the ERROR_METRICS argument.");
+        }
+
+        if (STRATIFIER_VALUE != null) {
+            errors.add("STRATIFIER_VALUE is a fake argument that is only there to show what are the different Stratification options. Please use it within the STRATIFIER_VALUE argument.");
+        }
+
+        final String[] superValidation = super.customCommandLineValidation();
+        if (superValidation != null) errors.addAll(Arrays.asList(superValidation));
+        if (!errors.isEmpty()) {
+            return errors.toArray(new String[errors.size()]);
+        } else {
+            return null;
+        }
+    }
+
+    private final Log log = Log.getInstance(CollectBamErrorMetrics.class);
+
+    @Override
+    protected int doWork() {
+
+        final Random random = new Random(42);
+
+        final ProgressLogger progressLogger = new ProgressLogger(log, 100000);
+        long nTotalLoci = 0;
+        long nSkippedLoci = 0;
+        long nProcessedLoci = 0;
+
+        IOUtil.assertFileIsReadable(INPUT);
+        IOUtil.assertFilesAreReadable(INTERVALS);
```

Comment:
Assert REFERENCE_SEQUENCE is readable?  How about the VCF? Then you don't need the ugly try/catch below.